R/mock2.R

```diff
@@ -0,0 +1,96 @@
+#' Mocking tools
+#'
+#' @description
+#' `r lifecycle::badge("experimental")`
+#'
+#' These functions represent a second attempt at bringing mocking to testthat,
+#' incorporating what we've learned from mockr and mockery There are
+#' two styles of mocking:
+#'
+#' * `local_mocked_bindings()` and `local_mocked_bindings()` work by temporarily
+#'   changing variable bindings in the package namespace. It affects the
+#'   operation of every function in the current package, but only the current
+#'   package.
+#'
+#' * `with_mocked_env()` modifies a function, inserting a new environment in the
```

Comment:
Is there a reason why we don't have `local_mocked_env()`?

Reply:
Probably this variant should be called `set_mocked_env()` since it's not really a `with_` function (hence why it doesn't have a `local_` variant).